packages/cli/src/metadataGeneration/typeResolver.ts

```diff
@@ -471,27 +471,50 @@ export class TypeResolver {
     return nodes;
   }
 
+  private hasFlag(type: ts.Symbol | ts.Declaration, flag) {
+    return (type.flags & flag) === flag;
+  }
+
   private getEnumerateType(typeName: ts.EntityName): Tsoa.RefEnumType | undefined {
     const enumName = (typeName as ts.Identifier).text;
-    let enumNodes = this.current.nodes.filter(node => node.kind === ts.SyntaxKind.EnumDeclaration).filter(node => (node as any).name.text === enumName);
+
+    const symbol = this.getSymbolAtLocation(typeName);
+
+    // resolve value
+    let declaredType = (this.current.typeChecker.getDeclaredTypeOfSymbol(symbol)?.symbol || symbol) as ts.Symbol & { parent?: ts.Symbol };
+
+    // if we are a EnumMember, return parent instead (this happens if a enum has only one entry, not quite sure why though...)
+    if (this.hasFlag(declaredType, ts.SymbolFlags.EnumMember) && declaredType.parent?.valueDeclaration.kind === ts.SyntaxKind.EnumDeclaration) {
+      declaredType = declaredType.parent;
+    }
+
+    const allNodes = declaredType.getDeclarations() as ts.DeclarationStatement[];
+
+    let enumNodes = allNodes.filter(node => {
+      if (!this.nodeIsUsable(node) || !this.current.IsExportedNode(node)) {
+        return false;
+      }
+
+      return node.kind === ts.SyntaxKind.EnumDeclaration && node.name?.text === enumName;
+    }) as ts.EnumDeclaration[];
```

Comment:
```suggestion
    let enumNodes = allNodes.filter((node): node is ts.EnumDeclaration => {
      if (!this.nodeIsUsable(node) || !this.current.IsExportedNode(node)) {
        return false;
      }

      return node.kind === ts.SyntaxKind.EnumDeclaration && node.name?.text === enumName;
    });
```

Reply:
mhm I know this is subjective, but I see my versoin more correct, as the node that is passed into the filter function is not for sure a ts.EnumDeclaration yet, we only can say so after the node.kind === ts.SyntaxKind.EnumDeclaration check, but I will look into if I can find some pre defined guard for it

Reply:
... I didn't "type" the incoming node,.
That is still its orginal type defined by the inference of the array.

I put the type on the return type as a guard function. 

You can do the same with functions that throw errors if something is not correct. In that case it would be
```ts
assertNode(node): asserts node is ts.EnumDeclation {
  ....
}
```

Reply:
ah right, sorry now I get your idea :-)!